openpype/modules/ftrack/plugins/publish/integrate_ftrack_status.py

```diff
@@ -0,0 +1,435 @@
+        return super(IntegrateFtrackFarmStatus, self).is_valid_instance(
+            context, instance)
+
+    def get_status_profiles(self):
+        if self.status_profiles is None:
+            profiles = copy.deepcopy(self.farm_status_profiles)
+            for profile in profiles:
+                profile["host_names"] = profile.pop("hosts")
+                profile["subset_names"] = profile.pop("subsets")
+            self.status_profiles = profiles
+        return self.status_profiles
+
+
+class IntegrateFtrackLocalStatus(IntegrateFtrackStatusBase):
+    """Collect task status names for instances that are published locally.
+
+    Instance which has set "farm" key in data to 'True' is considered as will
+    be rendered on farm thus it's status should be changed.
+
+    Requirements:
+        projectName (str): Name of the project.
+        hostName (str): Name of the host.
+        ftrackSession (ftrack_api.Session): Prepared ftrack session.
+        ftrackTaskStatuses (dict[str, list[Any]]): Dictionary of available
+            task statuses on project by task type id.
+        ftrackStatusByTaskId (dict[str, str]): Empty dictionary of task
+            statuses by task id. Status on task can be set only once.
+            Value should be a name of status.
+    """
+
+    order = IntegrateFtrackFarmStatus.order + 0.001
+    label = "Ftrack Task Status Local Publish"
+    active = True
+    targets = ["local"]
+    settings_key = "ftrack_task_status_local_publish"
+
+    def is_valid_instance(self, context, instance):
+        if instance.data.get("farm"):
+            self.log.debug("{} Will be rendered on farm.".format(
+                instance.data["subset"]
+            ))
+            return False
+        return super(IntegrateFtrackLocalStatus, self).is_valid_instance(
+            context, instance)
+
+
+class IntegrateFtrackOnFarmStatus(IntegrateFtrackStatusBase):
+    """Collect task status names for instances that are published on farm.
+
+    Requirements:
+        projectName (str): Name of the project.
+        hostName (str): Name of the host.
+        ftrackSession (ftrack_api.Session): Prepared ftrack session.
+        ftrackTaskStatuses (dict[str, list[Any]]): Dictionary of available
+            task statuses on project by task type id.
+        ftrackStatusByTaskId (dict[str, str]): Empty dictionary of task
+            statuses by task id. Status on task can be set only once.
+            Value should be a name of status.
+    """
+
+    order = IntegrateFtrackLocalStatus.order + 0.001
+    label = "Ftrack Task Status On Farm Status"
+    active = True
+    targets = ["farm"]
+    settings_key = "ftrack_task_status_on_farm_publish"
+
+
+class IntegrateFtrackTaskStatus(pyblish.api.ContextPlugin):
+    # Use order of Integrate Ftrack Api plugin and offset it before or after
+    base_order = pyblish.api.IntegratorOrder + 0.499
+    # By default is after Integrate Ftrack Api
+    order = base_order + 0.001
```

Comment:
IntegratorOrder + 0.5 means it's basically not an integrator anymore right? It's in the next order range (which is not defined in Pyblish itself, but I think that's referred to as "Other" range.

Reply:
```suggestion
    # Use order of Integrate Ftrack Api plugin and offset it before or after
    base_order = pyblish.api.IntegratorOrder + 0.499
    # By default is after Integrate Ftrack Api
    order = base_order + 0.0001
```